doc/classes/String.xml

```diff
@@ -232,30 +236,49 @@
 			</argument>
 			<description>
 				Converts a string containing a hexadecimal number into a decimal integer. If [code]with_prefix[/code] is [code]true[/code], the hexadecimal string should start with the [code]0x[/code] prefix, otherwise [code]0[/code] is returned.
-				[codeblock]
-				print("0xff".hex_to_int()) # Print "255"
-				print("ab".hex_to_int(false)) # Print "171"
-				[/codeblock]
+				[codeblocks]
+				[gdscript]
+				print("0xff".hex_to_int()) # Prints "255".
+				print("ab".hex_to_int(false)) # Prints "171".
+				[/gdscript]
+				[csharp]
+				GD.Print("0Xff".HexToInt()); // Prints "255".
+				// C# doesn't have `HexToInt(bool)`.
+				// GD.Print("ab".HexToInt(false)); // Prints "171".
+				[/csharp]
+				[/codeblocks]
 			</description>
 		</method>
 		<method name="http_escape">
 			<return type="String">
 			</return>
 			<description>
 				Escapes (encodes) a string to URL friendly format. Also referred to as 'URL encode'.
-				[codeblock]
+				[codeblocks]
+				[gdscript]
 				print("https://example.org/?escaped=" + "Godot Engine:'docs'".http_escape())
-				[/codeblock]
+				[/gdscript]
+				[csharp]
+				// We don't have HttpEscape in C#?
+				//GD.Print("https://example.org/?escaped=" + "Godot Engine:'docs'".HttpEscape());
```

Comment:
Another C# doesn't have it case.

Reply:
There is a native C# way to do this: `System.Net.WebUtility.UrlEncode` (and the inverse `System.Net.WebUtility.UrlDecode`).

However, in C# spaces are encoded as `+`, while in Godot spaces are encoded as `%20`. This seems to be a [hot debate](https://stackoverflow.com/questions/1634271/url-encoding-the-space-character-or-20). There is no option in C# to make it use `%20`, and `%20` seems to be broadly accepted as the better format. Both are valid URLs, and the GDScript-encoded version being decoded in C# is fine, but the C#-encoded version being decoded in GDScript is not valid because GDScript does not currently handle this alternate format. C# is able to decode both formats, but GDScript can't decode the `+` format.

One idea is to wrap around `System.Net.WebUtility.UrlEncode` by replacing `+` with `%20`, which would give the same behavior as GDScript. The `%20` format can be decoded everywhere.

Another idea is that we can tweak the decoding in GDScript to handle the `+` scheme. If that is done, both GDScript and C# will be able to decode from each other and themselves, even though they will encode different things. This does change the behavior if something expected `+` to be `+` and not `%2B`, but I don't think anyone uses this format.

We can actually do both of these things at the same time. If we do only the first one, then C# `string.HttpEscape` (or `HTTPEscape`?) will encode the same as GDScript, but then GDScript still won't handle the alternate format such as if `System.Net.WebUtility.UrlEncode` is used directly. If we only do the second option, then the decode methods would be intercompatible, but the encode methods would give different results. If we do both, then I think that solves all of the problems.